pkg/gohai/go.mod

```diff
@@ -2,7 +2,7 @@ module github.com/DataDog/datadog-agent/pkg/gohai
 
 // we don't want to just use the agent's go version because gohai might be used outside of it
 // eg. opentelemetry
-go 1.18
+go 1.20
```

Comment:
Gohai has to keep working with older versions, as explained in the comment above.
```suggestion
go 1.18
```

Reply:
So does gohai have to support older, unsupported by Google go versions?

Reply:
1.20 is the oldest supported version now

Reply:
I would say Gohai has to work on the oldest version between the datadog-agent and otel (no matter what the oldest supported version is), but I just checked and they updated to 1.20 four days ago so ignore my previous comment :smile: